packages/cloud_firestore/cloud_firestore_platform_interface/test/method_channel_tests/method_channel_query_test.dart

```diff
@@ -203,16 +203,18 @@ void main() {
     });
 
     group("snapshots()", () {
-      int handleId;
       final List<MethodCall> log = <MethodCall>[];
+
+      final String mockObserverId = 'QUERY1';
+
       setUp(() {
         log.clear();
-        handleId = nextMockHandleId;
 
         handleMethodCall((MethodCall call) async {
           log.add(call);
-          if (call.method == "Query#addSnapshotListener") {
-            await Future<void>.delayed(Duration.zero);
+          if (call.method == "Query#snapshots") {
+            handleQuerySnapshotsEventChannel(mockObserverId, log);
+            return Future.value(mockObserverId);
```

Comment:
Is there a reason for the `Future.value`? As we're in an async function, so we could do:

```suggestion
            return mockObserverId;
```